src/service/performance-impl.js

```diff
@@ -353,29 +411,19 @@ export class Performance {
       } else if (entry.entryType === 'layout-shift') {
         // Ignore layout shift that occurs within 500ms of user input, as it is
         // likely in response to the user's action.
-        // 1000 here is a magic number to prevent unbounded growth. We don't expect it to be reached.
-        if (!entry.hadRecentInput && this.layoutShifts_.length < 1000) {
-          this.layoutShifts_.push(entry);
+        if (!entry.hadRecentInput) {
+          this.tickNominalLayoutShiftScore_(entry);
+          // 1000 here is a magic number to prevent unbounded growth. We don't expect it to be reached.
+          if (this.layoutShiftsCumulative_.length < 1000) {
+            this.layoutShiftsCumulative_.push(entry);
+          }
         }
       } else if (entry.entryType === 'largest-contentful-paint') {
         this.largestContentfulPaint_ = entry.startTime;
         if (entry.element) {
-          const {tagName} = getOutermostAmpElement(entry.element);
-          if (tagName === 'IMG' || tagName === 'AMP-IMG') {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.image;
-          } else if (tagName === 'VIDEO' || tagName === 'AMP-VIDEO') {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.video;
-          } else if (tagName === 'AMP-CAROUSEL') {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.carousel;
-          } else if (tagName === 'AMP-BASE-CAROUSEL') {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.bcarousel;
-          } else if (tagName === 'AMP-AD') {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.ad;
-          } else {
-            this.largestContentfulPaintType_ = LCP_ELEMENT_TYPE.other;
-          }
+          this.largestContentfulPaint_ = getElementType(entry.element);
```

Comment:
Suggestion: make`getElementType(Element): ELEMENT_TYPE` into `getTagType(string?): ELEMENT_TYPE`, drop this whole conditional and  `this.largestContentfulPaint = getTagType(entry.element?.tagName)`, letting `getTagType` handle the `undefined` tagName as it already does

Reply:
Done.